packages/lit-analyzer/src/lib/analyze/parse/parse-dependencies/visit-dependencies.ts

```diff
@@ -158,8 +158,15 @@ function emitDirectModuleImportWithName(moduleSpecifier: string, node: Node, con
 		result = (context.program as any)["getResolvedModuleWithFailedLookupLocationsFromCache"](moduleSpecifier, fromSourceFile.fileName);
 	} else {
 		const cache = (context.program as MaybeModernProgram).getModuleResolutionCache?.();
+		let mode: tsModule.ModuleKind.CommonJS | tsModule.ModuleKind.ESNext | undefined = undefined; // tsModule.ModuleKind.ESNext;
+		if ((context.ts.isImportDeclaration(node) && !node.importClause?.isTypeOnly) || (context.ts.isExportDeclaration(node) && !node.isTypeOnly)) {
```

Comment:
```ts
!node.importClause?.isTypeOnly
```

Should this instead be:

```ts
(node.importClause && !node.importClause.isTypeOnly)
```

Because this import doesn't have an import clause, but it shouldn't be treated like a type only import:

```ts
import 'foo';
```

Hm, actually, looking at the docs for getModeForUsageLocation, I think we should just pass the node itself and not have any of these checks, including the typeOnly check:

```ts
/**
 * Calculates the final resolution mode for a given module reference node. This is generally the explicitly provided resolution mode, if
 * one exists, or the mode of the containing source file. (Excepting import=require, which is always commonjs, and dynamic import, which is always esm).
 * Notably, this function always returns `undefined` if the containing file has an `undefined` `impliedNodeFormat` - this field is only set when
 * `moduleResolution` is `node16`+.
 * @param file The file the import or import-like reference is contained within
 * @param usage The module reference string
 * @returns The final resolution mode of the import
 */
```

Reply:
Yeah I tried to do that without the checks but it broke in TS 4.8.3 while passed in later TS versions. Not sure why exactly.

But yeah this conditional I just copied from elsewhere in this file so I'm not sure its the right one exactly.

Reply:
How does it break in 4.8.3?